### PR TITLE
Add rollback CLI option

### PR DESCRIPTION
**Pull request by skazi0**

### CODE REVIEW

**Comment by skazi0** [comment]
I rebased and moved the code back to `connect` package. We can extract it to a separate one if needed in the future.

File: DIFFERENCES.md [hunk 1]
```diff
@@ -17,3 +17,9 @@
   Go version will handle this as "API error" while ruby version will fail to
   parse the response and handle this as "JSON error".
 - The Go HTTP client tries to reuse connections with keep-alive.
+- Docs for GET APIs call for URL-encoded query params. Ruby version sends a
```
Comment (djoreilly) [inline DIFFERENCES.md:20]: Is this related to rollback somehow?

File: connect/api.go [hunk 2]
```diff
@@ -146,6 +150,24 @@ func deregisterSystem() error {
 	return err
 }
 
+func synchronizeProducts(products []Product) ([]Product, error) {
```
Comment (djoreilly) [inline connect/api.go:153]: kinda long name, maybe syncProducts() or synchronize() like the Ruby version

File: suseconnect/usage.txt [hunk 3]
```diff
@@ -40,6 +40,9 @@ Manage subscriptions at https://scc.suse.com
         --write-config       Write options to config file at /etc/SUSEConnect.
         --cleanup            Remove old system credentials and all zypper
                              services installed by SUSEConnect.
+        --rollback           Revert the registration state in case of a failed
+                             migration.
+
```
Comment (djoreilly) [inline suseconnect/usage.txt:46]: The Ruby version has just one blank line before `Common options:`

**Review by djoreilly — CHANGES_REQUESTED** [review]
Just some small comments. Yes we can revisit the packaging later when needed.

File: DIFFERENCES.md (hunk 1 above)
Comment (skazi0) [reply DIFFERENCES.md:20]: I noticed this while comparing debug traces of old and new version. IIRC this was visible in `showProduct()` GET call.

File: connect/api.go (hunk 2 above)
Comment (skazi0) [reply connect/api.go:153]: `synchronize()` looks a bit more generic to me. `syncProducts()` is fine.

File: suseconnect/usage.txt (hunk 3 above)
Comment (skazi0) [reply suseconnect/usage.txt]: good catch! fixed.

File: connect/api.go (hunk 2 above)
Comment (skazi0) [reply connect/api.go]: done

File: DIFFERENCES.md (hunk 1 above)
Comment (djoreilly) [reply DIFFERENCES.md:20]: ok, yes I checked and showProduct() is the only api call that uses toQuery(). I think we should specify the difference is between Go showProduct() and Ruby show_product(), because as it stands the reader would have to go compare the API doc, Go code and Ruby code, to find which call(s) this is about

File: DIFFERENCES.md [hunk 4]
```diff
@@ -17,3 +17,9 @@
...
+  JSON query in body (like for other verbs). Go implementation follows docs.
+- The connect.synchronizeProducts() returns deserialized slice of Product
```
Comment (djoreilly) [inline DIFFERENCES.md:22]: connect.syncProducts() - but not sure that this is a difference we need to document. The function is not exported.

**Review by djoreilly — CHANGES_REQUESTED** [review]

File: DIFFERENCES.md (hunk 1 above)
Comment (skazi0) [reply DIFFERENCES.md:20]: added note about `showProduct()`

File: DIFFERENCES.md (hunk 4 above)
Comment (skazi0) [reply DIFFERENCES.md]: I'd rather keep it for developers as a sign of by-design change in behavior.

**Review by djoreilly — APPROVED** [review]
lgtm
